include/qx/Core.hpp

```diff
@@ -42,7 +42,7 @@ template <std::size_t N> class DenseUnitaryMatrix {
     explicit constexpr DenseUnitaryMatrix(Matrix const &m)
         : DenseUnitaryMatrix(m, true) {}
 
-    [[nodiscard]] inline constexpr std::complex<double> at(std::size_t i, std::size_t j) const {
+    [[nodiscard]] inline constexpr const std::complex<double>& at(std::size_t i, std::size_t j) const {
```

Comment:
Hey Pablo, it's not a performance issue but more of being consistent with STL implementations, i.e., STL data structures return a reference, or a const reference, for `at` (see [`vector`](https://en.cppreference.com/w/cpp/container/vector/at)'s for instance).

We could also add a non-const version of `at` but it seems it is not needed for the current code.

Reply:
I think they return a reference because it's better in the general case and they are polymorphic/templated.
It doesn't matter anyway I think. OK for me.